tests/phpunit/integration/Modules/AnalyticsTest.php

```diff
@@ -458,4 +461,50 @@ protected function get_module_with_screen() {
 	protected function get_module_with_settings() {
 		return new Analytics( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
 	}
+
+	public function test_parse_data_orderby() {
+		$analytics = new Analytics( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+
+		$reflected_parse_data_orderby_method = new ReflectionMethod( 'Google\Site_Kit\Modules\Analytics', 'parse_data_orderby' );
+		$reflected_parse_data_orderby_method->setAccessible( true );
+
+		// When there is no orderby in the request.
+		$result = $reflected_parse_data_orderby_method->invoke( $analytics, new Data_Request( 'GET', null, null, 'report', array() ) );
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEmpty( $result );
+
+		// When a single order object is used.
+		$order  = array(
+			'fieldName' => 'views',
+			'sortOrder' => 'ASCENDING',
+		);
+		$result = $reflected_parse_data_orderby_method->invoke( $analytics, new Data_Request( 'GET', null, null, 'report', array( 'orderby' => $order ) ) );
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEquals( 1, count( $result ) );
+		$this->assertTrue( $result[0] instanceof Google_Service_AnalyticsReporting_OrderBy );
+		$this->assertEquals( $order['fieldName'], $result[0]->getFieldName() );
+		$this->assertEquals( $order['sortOrder'], $result[0]->getSortOrder() );
+
+		// When multiple orders are passed.
+		$orders = array(
+			array(
+				'fieldName' => 'pages',
+				'sortOrder' => 'DESCENDING',
+			),
+			array(
+				'fieldName' => 'sessions',
+				'sortOrder' => 'ASCENDING',
+			),
+		);
+		$result = $reflected_parse_data_orderby_method->invoke( $analytics, new Data_Request( 'GET', null, null, 'report', array( 'orderby' => $orders ) ) );
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEquals( 2, count( $result ) );
+		$this->assertTrue( $result[0] instanceof Google_Service_AnalyticsReporting_OrderBy );
+		$this->assertEquals( $orders[0]['fieldName'], $result[0]->getFieldName() );
+		$this->assertEquals( $orders[0]['sortOrder'], $result[0]->getSortOrder() );
+		$this->assertTrue( $result[1] instanceof Google_Service_AnalyticsReporting_OrderBy );
+		$this->assertEquals( $orders[1]['fieldName'], $result[1]->getFieldName() );
+		$this->assertEquals( $orders[1]['sortOrder'], $result[1]->getSortOrder() );
+	}
```

Comment:
Let's also test that an orderby is not returned if it's `fieldName` is empty. Pending my previous suggestion, we should do the same if the `sortOrder` is also empty (currently this would never happen though since it has the default).

Reply:
I have added a test to check the case when `fieldName` is empty. Will udpate it if Felix agrees with your suggestion.

Reply:
Updated.